### PR TITLE
MCOL-5191 Add MCV statistics.

### PR DESCRIPTION
**Pull request by denis0x0D**
This patch adds:
1. Initial version of random sampling.
2. Initial version of MCV statistics.

### CODE REVIEW

File: primitives/primproc/sqlfrontsessionthread.cpp [hunk 1]
```diff
@@ -238,19 +239,23 @@ namespace exemgr
...
-    msgHandler.stop();
-
-    auto outRG = (static_cast<joblist::TupleJobList*>(jl.get()))->getOutputRowGroup();
 
-    if (caep.traceOn())
-      std::cout << "Row count " << rowCount << std::endl;
-
-    // Process `RowGroup`, increase an epoch and save statistics to the file.
+    // Seemls like this a legacy parameter, not really needed.
+    const uint32_t dummyTableOid = 100;
```
Comment (drrtuy) [inline primitives/primproc/sqlfrontsessionthread.cpp:245]: It is indeed. Its semantics was important back in Columnstore prior to MCS 1.4.

File: utils/common/statistics.cpp [hunk 2]
```diff
@@ -31,61 +32,135 @@ using namespace logging;
...
+    // MCV statistics.
+    std::vector<pair<uint64_t, uint32_t>> mcvList(columnMCV.begin(), columnMCV.end());
+    std::sort(mcvList.begin(), mcvList.end(),
+              [](const std::pair<uint64_t, uint32_t>& a, const std::pair<uint64_t, uint32_t>& b) {
+                return a.second > b.second;
+              });
+
+    // 200 buckets as Microsoft does.
+    const auto mcvSize = std::min(columnMCV.size(), static_cast<uint64_t>(200));
+    mcv[oid] = std::unordered_map<uint64_t, uint32_t>(mcvList.begin(), mcvList.begin() + mcvSize);
```
Comment (drrtuy) [inline utils/common/statistics.cpp:131]: I would name a type for readability.

**Comment by drrtuy** [comment]
I missed where how exactly do you treat/filter out char/varchar/text dict columns?

**Comment by denis0x0D** [comment]
@drrtuy the char/varchar is not supported because it was not supported in the original pipeline, that means current analyze table command does not generate "job steps" to read the dictionary data. Probably that's what needs to be updated in analyze command.

**Review by drrtuy — APPROVED** [review]

**Pull request merged**